cattr/multistrategy_dispatch.py

```diff
@@ -0,0 +1,53 @@
+from .function_dispatch import FunctionDispatch
+from ._compat import singledispatch
+
+
+class _DispatchNotFound(object):
+    """ a dummy object to help signify a dispatch not found """
+    pass
+
+
+class MultiStrategyDispatch(object):
+    """
+    MultiStrategyDispatch uses a
+    combination of FunctionDispatch and singledispatch.
+
+    singledispatch is attempted first. If nothing is
```

Comment:
Just wanted to note here: we discussed using FunctionDispatch, then falling back to SingleDispatch, but I think that won't work to resolve subclasses: registering a subclass of an iterable, then having the functiondispatch pick that up when determined as a subclass for typing.List and runs iterable_seq.

So switched the order: singledispatch is attempted, then FunctionDispatch as a fallback.